browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -56,16 +60,88 @@ void OnShouldBlockAdResult(const ResponseCallback& next_callback,
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
   if (ctx->blocked_by == kAdBlocked) {
     brave_shields::DispatchBlockedEvent(
-        ctx->request_url,
-        ctx->render_frame_id, ctx->render_process_id, ctx->frame_tree_node_id,
-        brave_shields::kAds);
+        ctx->request_url, ctx->render_frame_id, ctx->render_process_id,
+        ctx->frame_tree_node_id, brave_shields::kAds);
   }
   next_callback.Run();
 }
 
-void OnBeforeURLRequestAdBlockTP(
-    const ResponseCallback& next_callback,
-    std::shared_ptr<BraveRequestInfo> ctx) {
+void OnGetCnameResult(const ResponseCallback& next_callback,
+                      std::shared_ptr<BraveRequestInfo> ctx,
+                      const base::Optional<std::string> cname) {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
+  scoped_refptr<base::SequencedTaskRunner> task_runner =
```

Comment:
This isn't safe because AdblockCnameResolveHostClient is not owned by anything and so it could outlive anything including the ad_block_service

Reply:
you could pass the task runner itself into the `AdblockCnameResolveHostClient` constructor since it's ref counted